src/game/Entities/Unit.cpp

```diff
@@ -12605,11 +12605,11 @@ Unit* Unit::TakePossessOf(SpellEntry const* spellEntry, SummonPropertiesEntry co
 
     const bool immunePC = IsImmuneToPlayer();
     if (possessed->IsImmuneToPlayer() != immunePC)
-        possessed->SetImmuneToPlayer(immunePC);
+        possessed->SetImmuneToPlayer(possessed->GetSettings().HasFlag(CreatureStaticFlags::IMMUNE_TO_PC) ? true : immunePC);
 
     const bool immuneNPC = IsImmuneToNPC();
     if (possessed->IsImmuneToNPC() != immuneNPC)
-        possessed->SetImmuneToNPC(immuneNPC);
+        possessed->SetImmuneToNPC(possessed->GetSettings().HasFlag(CreatureStaticFlags::IMMUNE_TO_NPC) ? true : immuneNPC);
```

Comment:
these are sus. should I just keep it in the spellscript as manual override?

Reply:
No, these are entirely separate, this code here shouldnt be altered. The code is literally "preserve what mob had before posses".

Reply:
in practice however this means that the mob from Dream Vision will not have IMMUNE_TO_PC and IMMUNE_TO_NPC unitflags, because with the code here as it was before it will be reset to whatever the player who summons the dream vision NPC has as flags (and the player doesn't get these flags)

Reply:
so either I need these changes here, or I have to separately set these flags in the SpellScript, because those are applied after the possession

Reply:
Removed the offending block entirely after examination. Rebase your thing and make sure the db data makes it there as intended